src/common/options/rgw.yaml.in

```diff
@@ -3999,3 +3999,11 @@ options:
   services:
   - rgw
   with_legacy: true
+- name: rgw_localize_read
+  type: bool
+  level: advanced
+  desc: Enable localization of read operations
+  default: false
```

Comment:
> why does this default to off? does it need to be an option at all?
> 

AFAIR RBD does not enable it by default, @idryomov any challenges in enabling it by default?

> it seems like local reads would always be preferred, which is why i'm confused that objecter doesn't try this by default
> 
> quoting @athanatos from the [Supporting localise Read for RGW](https://lists.ceph.io/hyperkitty/list/dev@ceph.io/message/QA7HDVVC5F5QKEOBTE2AXVGJA5EBKYN6/) mailing list thread,
> 
> > It would probably make sense to add support in the objecter for picking up certain flags (such as LOCALIZE_READS) from the pool config.
> 
> have we tried to pursue this?

Reply:
> AFAIR RBD does not enable it by default, @idryomov any challenges in enabling it by default?

Telling librados to localize reads wouldn't do anything useful unless `crush_location` for the client is set correctly.  AFAIK `crush_location` isn't set by default, at least in standalone Ceph.

One could probably do something with `crush_location_hook` there, but likely very specific to the environment.

Reply:
> @idryomov Currently I am tested in the k8s environment, so Rook can figure out crush_location for RGW when I set this option. Does `cephadm` have similar capabilities so that it can also be used in that deployment? I saw there is option in cephadm to set [crush_location](https://docs.ceph.com/en/quincy/cephadm/services/mon/#setting-crush-locations-for-monitors) for mon

I don't think so.  That option has a very specific purpose which isn't related to read locality -- it's there for stretch clusters where each monitor must have a location specified for tiebreaking.  @adk3798 would know for sure.

> @idryomov one more follow-up question assuming a local osd is running for the RGW server, the default location for `crush_location` will work right??

Since you also mention changing the default `crush_location` for RGW, I'm not sure.  This is something for you to test.